plan/statscache/statscache_test.go

```diff
@@ -33,44 +35,39 @@ var _ = Suite(&testStatsCacheSuite{})
 type testStatsCacheSuite struct{}
 
 func (s *testStatsCacheSuite) TestStatsCache(c *C) {
-	tblInfo := &model.TableInfo{
-		ID: 1,
-	}
-	columns := []*model.ColumnInfo{
-		{
-			ID:        2,
-			FieldType: *types.NewFieldType(mysql.TypeLonglong),
-		},
-	}
-	tblInfo.Columns = columns
-	ctx := mock.NewContext()
-	ctx.Store = kv.NewMockStorage()
-	statsTbl := GetStatisticsTableCache(ctx, tblInfo)
-	c.Check(len(statsTbl.Columns), Equals, 1)
+	store, do, err := newStoreWithBootstrap()
+	c.Assert(err, IsNil)
+	defer store.Close()
+	testKit := testkit.NewTestKit(c, store)
+	testKit.MustExec("use test")
+	testKit.MustExec("create table t (c1 int, c2 int)")
+	testKit.MustExec("insert into t values(1, 2)")
+	is := do.InfoSchema()
+	tbl, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	c.Assert(err, IsNil)
+	tableInfo := tbl.Meta()
+	statsTbl := statscache.GetStatisticsTableCache(tableInfo)
+	c.Assert(statsTbl.Pseudo, IsTrue)
+	testKit.MustExec("analyze table t")
+	statsTbl = statscache.GetStatisticsTableCache(tableInfo)
+	c.Assert(statsTbl.Pseudo, IsFalse)
+	testKit.MustExec("create index idx_t on t(c1)")
+	is = do.InfoSchema()
+	tbl, err = is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	c.Assert(err, IsNil)
+	tableInfo = tbl.Meta()
+	statsTbl = statscache.GetStatisticsTableCache(tableInfo)
+	c.Assert(statsTbl.Pseudo, IsTrue)
```

Comment:
When the schema has changed, we should not get a Pseudo table stats.
If the stats doesn't match the new schema, we should adapt the stats for the new schema.
Or we will keep using the pseudo stats for a long time.